Fix Graph Node close rect default position

SirQuartz opened the pull request:
The Graph Node has an optional close button that can be shown or hidden. When the close button is shown, it's positioned in a way where it visibly extends over the outer right edge of the node. This can be fixed by using theme overrides in the inspector, but by default, it's positioned awkwardly.

It would appear that the button is positioned so that its center aligns with the outer margin of the node when in reality, it should be the right edge of the button that's aligned with the right margin of the node.

Current default behavior:
<img width="602" alt="Graph Node" src="https://user-images.githubusercontent.com/62965063/220461550-b1cd8271-357b-4246-acf1-4b7612fc2144.png">

New behavior:
<img width="601" alt="Graph Node 2" src="https://user-images.githubusercontent.com/62965063/220461926-3e047559-e1fe-4b98-b63b-51cd6510a903.png">

This makes more sense as the default position than the current behavior. Having the close icon properly aligned means it won't have to be realigned by the user for every new node.

SirQuartz commented:
Upon further investigation, it would appear that there is a discrepancy between the visual shader graph nodes and the scene graph nodes in regards to the position of the close button

<img width="475" alt="Visual Shader" src="https://user-images.githubusercontent.com/62965063/220477155-84acf4bb-1fa5-40aa-9cd8-552f69f0c08e.png">

The in-engine visual shader nodes already have the close button properly aligned, so now my changes push it further to the left. Not sure why this difference exists in the first place.

Geometror reviewed (commented):
I think a proper fix would be changing the close_h_offset for the default theme. If you set all offsets to zero and use the most basic stylebox this is the position where the close button is drawn:
![grafik](https://user-images.githubusercontent.com/50084500/220585834-363a0de1-f128-4f7d-be26-3de21c0c61cc.png)
GraphNode is currently designed in a way that you are expected to position the button manually via theme constants, but that will work differently after the refactor.

SirQuartz committed:
Modify the default theme GraphNode close_h_offset

Modifies the close_h_offset to be 12 instead of 22. This better aligns the default position.

SirQuartz commented:
I agree.

Geometror reviewed (approved):
Looks good!

akien-mga commented:
Thanks!

YuriSizov commented:
Cherry-picked for 4.0.1.
